<a href="https://colab.research.google.com/github/panwarashu19/Deep-Learning-with-Python/blob/main/Handwritten_Digits_Classification.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Handwritten Digits Classification



We’ll use the MNIST
dataset, a classic in the machine-learning community, which has been around almost
as long as the field itself and has been intensively studied. It’s a set of 60,000 training
images, plus 10,000 test images, assembled by the National Institute of Standards and
Technology (the NIST in MNIST) in the 1980s.

In [2]:
#Loading the MNIST Dataset in Keras
from keras.datasets import mnist
(train_images, train_labels), (test_images, test_labels) = mnist.load_data()

11501568/11490434 [==============================] - 0s 0us/step


In [3]:
#The Network Architecture

from keras import models
from keras import layers

network=models.Sequential()
network.add(layers.Dense(512, activation='relu', input_shape=(28*28,)))
network.add(layers.Dense(10, activation='softmax'))

In [4]:
#The Compilation Step

network.compile(optimizer='rmsprop',
                loss='categorical_crossentropy',
                metrics=['accuracy'])

In [5]:
#Preparing the Image Data
train_images=train_images.reshape((60000, 28*28))
train_images=train_images.astype('float32')/255

test_images=test_images.reshape((10000, 28*28))
test_images=test_images.astype('float32')/255

In [9]:
#Preparing the Labels
from tensorflow.keras.utils import to_categorical
train_labels=to_categorical(train_labels)
test_labels=to_categorical(test_labels)

In [11]:
#Train the Model

network.fit(train_images, train_labels, epochs=5, batch_size=128)

Epoch 1/5
469/469 [==============================] - 17s 8ms/step - loss: 0.4315 - accuracy: 0.8758
Epoch 2/5
469/469 [==============================] - 4s 8ms/step - loss: 0.1139 - accuracy: 0.9661
Epoch 3/5
469/469 [==============================] - 4s 8ms/step - loss: 0.0721 - accuracy: 0.9788
Epoch 4/5
469/469 [==============================] - 4s 8ms/step - loss: 0.0504 - accuracy: 0.9843
Epoch 5/5
469/469 [==============================] - 4s 8ms/step - loss: 0.0362 - accuracy: 0.9898


Epoch 1/5
469/469 [==============================] - 4s 8ms/step - loss: 0.0294 - accuracy: 0.9913
Epoch 2/5
469/469 [==============================] - 4s 8ms/step - loss: 0.0226 - accuracy: 0.9933
Epoch 3/5
469/469 [==============================] - 4s 8ms/step - loss: 0.0172 - accuracy: 0.9951
Epoch 4/5
469/469 [==============================] - 4s 8ms/step - loss: 0.0128 - accuracy: 0.9963
Epoch 5/5
469/469 [==============================] - 4s 8ms/step - loss: 0.0102 - accuracy: 0.9972


 We quickly reach an accuracy of 0.9972 (99.72%) on the training data. Now let’s
check that the model performs well on the test set, too:

In [12]:
test_loss, test_acc=network.evaluate(test_images, test_labels)

print("Test Accuracy: ", test_acc)

313/313 [==============================] - 1s 2ms/step - loss: 0.0696 - accuracy: 0.9807
Test Accuracy:  0.9807000160217285
